### PR TITLE
Consistent method naming

### PR DESCRIPTION
**Pull request by Armando-Ibarra**
-Sources - renamed `requestAsync` method name to `createSepaSource`
-Instruments - renamed methods to delete `Instrument` word as is redundant over the class name
-PaymentsLinks - renamed `getAsync & createAsync` method name to `getPaymentLink & createPaymentLink`
-HostedPayments - renamed `createAsync & get` method name to `createHostedPaymentsPageSession & getHostedPaymentsPageDetails`

-Added missing validations for parameters on some clients

### CODE REVIEW

**Comment by Armando-Ibarra** [comment]
This PR also contains some re wrote on Sources Client, to make it equals to others SDK's

File: src/main/java/com/checkout/payments/hosted/HostedPaymentsClient.java [hunk 1]
```diff
@@ -4,8 +4,8 @@
 
 public interface HostedPaymentsClient {
 
-    CompletableFuture<HostedPaymentResponse> createAsync(HostedPaymentRequest hostedPaymentRequest);
+    CompletableFuture<HostedPaymentResponse> createHostedPaymentsPageSession(HostedPaymentRequest hostedPaymentRequest);
```
Comment (rsaestrela) [inline src/main/java/com/checkout/payments/hosted/HostedPaymentsClient.java:7]: Great stuff!

File: src/test/java/com/checkout/sources/SourcesClientImplTest.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                eq(request), isNull()))
+                .thenReturn(CompletableFuture.completedFuture(response));
+
+        final CompletableFuture<SepaSourceResponse> future = client.createSepaSource(request);
+
+        assertNotNull(future.get());
+        assertEquals(response, future.get());
+
+    }
+
```
Comment (rsaestrela) [inline src/test/java/com/checkout/sources/SourcesClientImplTest.java:64]: Extra line

**Review by rsaestrela — APPROVED** [review]
Nice work!

**Pull request merged**